components/service/glean/src/main/java/mozilla/components/service/glean/Glean.kt

```diff
@@ -381,45 +385,39 @@ open class GleanInternalAPI internal constructor () {
      * @param pingNames List of pings to send.
      * @return true if any pings had content and were queued for uploading
      */
-    internal fun sendPingsInternal(pingNames: List<String>): Boolean {
+    internal fun sendPingsInternalAsync(pingNames: List<String>): Deferred<Boolean> {
```

Comment:
I think I agree with @mdboom on the fact that we never use the result of this function.  If I remember correctly, it was originally used to determine whether or not to enqueue the Worker in WorkManager but a lot of how that works has changed since then.  Since we wouldn't have a return value anymore, much of the `Deferred` stuff could be done away with and all of the `@Suppress("DeferredResultUnused")` linter suppressions as well.  Unless you see a use-case for using the returned Boolean in some way?

Reply:
If we really wanted to return something and add functionality to the public API, I think I'd much rather see a completion handler that ran after the pings were _actually_ uploaded, rather than returning a Boolean indicating that "yeah, I plan on sending some pings later...".  Otherwise, I'd vote to remove the return handler and Deferred stuff as it adds complexity.

Reply:
Good point on removing the returned `Boolean`.